kie-server-parent/kie-server-tests/kie-server-integ-tests-optaplanner/src/test/java/org/kie/server/integrationtests/optaplanner/jms/OptaPlannerJmsResponseHandlerIntegrationTest.java

```diff
@@ -98,43 +99,44 @@ public void testSolverWithAsyncResponseHandler() throws Exception {
         ResponseHandler asyncResponseHandler = new AsyncResponseHandler(responseCallback);
         solverClient.setResponseHandler(asyncResponseHandler);
 
-        ServiceResponse<?> response = solverClient.createSolver(CONTAINER_1_ID, SOLVER_1_ID, SOLVER_1_CONFIG);
+        SolverInstance response = solverClient.createSolver(CONTAINER_1_ID, SOLVER_1_ID, SOLVER_1_CONFIG);
         assertThat(response);
         SolverInstance solver = responseCallback.get(SolverInstance.class);
         assertThat(solver).isNotNull();
         assertThat(solver.getContainerId()).isEqualTo(CONTAINER_1_ID);
         assertThat(solver.getSolverId()).isEqualTo(SOLVER_1_ID);
 
-        response = solverClient.getSolvers(CONTAINER_1_ID);
-        assertThat(response);
+        List<SolverInstance> solverInstanceList = solverClient.getSolvers(CONTAINER_1_ID);
+        assertThat(solverInstanceList);
         SolverInstanceList solverList = responseCallback.get(SolverInstanceList.class);
         assertThat(solverList).isNotNull();
         assertThat(solverList.getContainers()).isNotNull().isNotEmpty().hasSize(1);
         solver = solverList.getContainers().get(0);
         assertThat(solver.getSolverId()).isEqualTo(SOLVER_1_ID);
         assertThat(solver.getStatus()).isEqualTo(SolverInstance.SolverStatus.NOT_SOLVING);
 
-        solver = new SolverInstance();
-        solver.setStatus(SolverInstance.SolverStatus.SOLVING);
-        solver.setPlanningProblem(loadPlanningProblem(5, 15));
-        response = solverClient.updateSolverState(CONTAINER_1_ID, SOLVER_1_ID, solver);
+        solverClient.solvePlanningProblem(CONTAINER_1_ID, SOLVER_1_ID, loadPlanningProblem(5, 15));
+        responseCallback.get(Void.class);
+
+        response = solverClient.getSolver(CONTAINER_1_ID, SOLVER_1_ID);
         assertThat(response);
         solver = responseCallback.get(SolverInstance.class);
         assertThat(solver.getSolverId()).isEqualTo(SOLVER_1_ID);
         assertThat(solver.getStatus()).isEqualTo(SolverInstance.SolverStatus.SOLVING);
 
-        response = solverClient.getSolverState(CONTAINER_1_ID, SOLVER_1_ID);
+        solverClient.terminateSolverEarly(CONTAINER_1_ID, SOLVER_1_ID);
+        responseCallback.get(Void.class);
```

Comment:
Can we extract that one liner in a method in that unit test (or a parent unit test or a TestUtil) named something like makeSureAllResultsAreConsumed() - that way it's clear what the purpose is when reading the unit test (and the details of have that's done is less relevant).